controller/PlatformAdmin.php

```diff
@@ -32,86 +36,133 @@
-        $myForm = $formContainer->getForm();
-
-        if($myForm->isSubmited()){
-            if($myForm->isValid()){
+        $this->setData('formTitle', __('Edit Instance'));
+        $this->saveInstance($this->getCurrentInstance());
+    }
 
-                // For treeBox component we need to add slashes for rdf value before saving
-                $properties = $publishingService->addSlashes($myForm->getValues());
-                $instance = $this->createInstance(array($clazz), $properties);
+    public function addInstanceForm()
+    {
+        $this->setData('formTitle', __('Create instance'));
+        $this->saveInstance();
+    }
 
-                $this->setData('message', __($instance->getLabel().' created'));
-                $this->setData('reload', true);
+    /**
+     * @throws \core_kernel_persistence_Exception
+     * @throws \tao_models_classes_MissingRequestParameterException
+     */
+    public function authConfiguration()
+    {
+        $instance = $this->getCurrentInstance();
+        $authType = $instance->getOnePropertyValue($this->getProperty(PlatformService::PROPERTY_AUTH_TYPE));
+
+        /** @var common_ext_ExtensionsManager $extensionManagerService */
+        $extensionManagerService = $this->getServiceLocator()->get(common_ext_ExtensionsManager::SERVICE_ID);
+        $authClass = $this->getClass(BasicAuth::CLASS_BASIC_AUTH);
+
+        $credentials = [
+            'type' => BasicAuth::CLASS_BASIC_AUTH,
+            'allowedTypes' => [BasicAuth::CLASS_BASIC_AUTH => [
+                'label' => $authClass->getLabel(),
+                'selected' => false,
+            ]],
+            'login' => '',
+            'password' => '',
+        ];
+
+        if ($authType) {
+            switch ($authType->getUri()) {
+                case BasicAuth::CLASS_BASIC_AUTH:
+                    $credentials['login'] = $instance->getOnePropertyValue($this->getProperty(BasicAuth::LOGIN))->literal;
+                    $credentials['password'] = $instance->getOnePropertyValue($this->getProperty(BasicAuth::PASSWORD))->literal;
+                    $credentials['allowedTypes'][BasicAuth::CLASS_BASIC_AUTH]['selected'] = true;
+                    break;
+                case 'oauth2':
+                    break;
             }
         }
 
-        $actionsElementClass = \tao_helpers_Uri::encode(PublishingService::PUBLISH_ACTIONS);
-        $actionsElement = $myForm->getElement($actionsElementClass);
-        $actionsElement->setOptions($publishingService->getPublishingActions());
-
-        $myForm->removeElement($actionsElement);
-        $myForm->addElement($actionsElement);
-
-        $this->setData('formTitle', __('Create instance of ').$clazz->getLabel());
-        $this->setData('myForm', $myForm->render());
+        // oAuth 1.0 // needs to be changed to taoOauth 2.0
+        if ($extensionManagerService->isInstalled('taoOauth')) {
```

Comment:
Could u retrieve this list from config please. taoOauth will register a new model instead of checking if it's installed